lib/api/src/externals/function.rs

```diff
@@ -727,19 +808,27 @@ impl VMDynamicFunction for VMDynamicFunctionWithoutEnv {
     }
 }
 
-#[repr(C)]
-pub(crate) struct VMDynamicFunctionWithEnv<Env>
+pub(crate) struct DynamicFunctionWithEnv<Env>
 where
     Env: Sized + 'static,
 {
-    // This field _must_ come first in this struct.
-    env: Box<Env>,
```

Comment:
Why `env` is no longer required to be at first position?

Reply:
The reason before was due to a bad hack where we'd double deref a pointer ignoring the fact that it was in a struct.

We now have each function that is called on the `env` traverse things correctly, so for example we create a wrapper function where we walk through the types and find what we need.  So if we need to operate on the `Box<env>` itself, we can find it in each function.  I think this is a simpler solution because then all the functions always just work on the `env`, we had an issue where the clone and drop function needed to be called on the top level the wrapper dynamic func context thing but the initialize function needed to be called on the inner env.  This fixes that by making the thing each function gets consistent.